lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -234,6 +271,48 @@ public static TokenDetails fromJSON(JsonObject json) {
 			if(timestamp > 0) params.add(new Param("timestamp", String.valueOf(timestamp)));
 			return params;
 		}
+
+		/**
+		 * Internal, when TokenParams are provided, the values of each attribute are not merged,
+		 * but instead replace all corresponding values, even when null.
+		 *
+		 * should be used for:
+		 * - {@link Auth#requestToken(TokenParams, AuthOptions)} (Spec:RSA8e),
+		 * - {@link Auth#createTokenRequest(AuthOptions, TokenParams)} (Spec:RSA9h),
```

Comment:
The method signature is wrong, it's `TokenParams, AuthOptions` for all methods.  It's in the spec you're referencing.